app.test.js

```diff
@@ -0,0 +1,83 @@
+const request = require('supertest');
+const app = require('./app');
+const regeneratorRuntime = require('regenerator-runtime');
+
+const environment = process.env.NODE_ENV || 'development';
+const configuration = require('./knexfile')[environment];
+const database = require('knex')(configuration);
+
+describe('Server', () => {
+
+  beforeEach( async () => {
+    await database.seed.run();
+  });
+
+  describe('init', () => {
+    it('should return a 200 status', async () => {
+      const res = await request(app).get('/');
+      expect(res.status).toBe(200);
+    });
+  });
+
+  describe('GET /projects', () => {
+    it('should return a 200 and all of the projects', async () => {
+      const expectedProjects = await database('projects').select();
+      const response = await request(app).get('/projects');
+      const projects = response.body;
+      expect(response.status).toBe(200);
+      expect(JSON.stringify(projects)).toEqual(JSON.stringify(expectedProjects));
+    });
+  });
+
+  describe('GET /palettes', () => {
+    it('should return a 200 and all of the palettes', async () => {
+      const expectedPalettes = await database('palettes').select();
+      const response = await request(app).get('/palettes');
+      const palettes = response.body;
+      expect(response.status).toBe(200);
+      expect(JSON.stringify(palettes)).toEqual(JSON.stringify(expectedPalettes));
```

Comment:
I'm also wondering if the need to use JSON.stringify is an artifact of the other syntax problems.